index.js

```diff
@@ -537,11 +538,9 @@ class HtmlWebpackPlugin {
      * if a path publicPath is set in the current webpack config use it otherwise
      * fallback to a relative path
      */
-    const webpackPublicPath = webpackMajorVersion === 4
-      ? compilation.mainTemplate.getPublicPath({ hash: compilationHash })
-      : compilation.getAssetPath(compilation.outputOptions.publicPath, { hash: compilationHash });
+    const webpackPublicPath = getPublicPath(compilation, { hash: compilationHash });
 
-    const isPublicPathDefined = webpackMajorVersion === 4
+    const isPublicPathDefined = isWebpack4()
```

Comment:
What would be the improvement here?

Reply:
Because `isWebpack4` gets the version of the `webpack.config.js` from which this plugin was loaded. 

https://github.com/jantimon/html-webpack-plugin/blob/f63b166cfb0fab301450dc6d314ff9f8aa530264/lib/webpack-polyfill.js#L40 

https://github.com/rubeniskov/webpack-import-plugins#bonus-track

![image](https://user-images.githubusercontent.com/6261914/98536539-4563d800-2288-11eb-8faa-ddf82dcadc56.png)